core/parser/src/lexer/regex.rs

```diff
@@ -139,7 +149,7 @@ impl<R> Tokenizer<R> for RegexLiteral {
             }
         }
 
-        if let Err(error) = Regex::from_unicode(body.into_iter(), flags_str) {
+        if let Err(error) = Regex::from_unicode(body.into_iter(), flags) {
```

Comment:
not related to this PR, but it would be good to cache the result of this and reuse it when we actually come to need it, but thats outside the scope of optimizing the flag parsing